strutil/quantity/quantity.go

```diff
@@ -200,3 +201,262 @@ func FormatDuration(dt float64) string {
 
 	return FormatAmount(uint64(dt), 4) + years
 }
+
+// Duration is represented in nano seconds.
+type Duration uint64
+
+const (
+	NSecond Duration = 1
+	USecond Duration = 1000
+	MSecond Duration = 1000000
+	Second  Duration = 1000000000
+	Minute  Duration = (60 * Second)
+	Hour    Duration = (60 * Minute)
+	Day     Duration = (24 * Hour)
+	Year    Duration = (1461 * Day / 4)
+)
+
+// Maximum number of places (units of time) to render starting with years.
+// In the Compact case, only the first two places will be rendered,
+// discarding the rest. This allows for a more predictable width (at the
+// expense of accuracy).
+type Places uint32
+
+const (
+	ShowVerbose Places = 8
+	ShowCompact Places = 2
+)
+
+// Space delimit options
+type SpaceMode bool
+
+const (
+	SpaceOff SpaceMode = false
+	SpaceOn  SpaceMode = true
+)
+
+// The minimum Duration parameter allows the rendering to discard the units
+// of time less than the minimum. However, depending on what we are
+// rendering, we either need to peform a ceiling, floor or rounding
+// operation with the remainder below the minimum.
+type RenderMode uint32
+
+const (
+	TimeLeft RenderMode = iota
+	TimePassed
+	TimeRounded
+)
+
+// FormatDurationGeneric formats time duration similar to systemd
+// format_timespan, but with options allowing for a more compact
+// arrangement, including shortened suffixes. The function is provided
+// with dt in seconds (as in the output of time.Now().Seconds()). The 'min'
+// and 'max' Duration allows limiting the unit range rendered. Place values
+// less than 'min' will be processed according to RenderMode. RenderMode
+// allows rounding, flooring or ceiling to be applied to place value below
+// 'min'. Durations above the 'max' place value will result in "ages!".
+func FormatDurationGeneric(dt float64, min Duration, max Duration, count Places, mode RenderMode, space SpaceMode) string {
+	var units = map[Duration]string{
+		Year:    "y",
+		Day:     "d",
+		Hour:    "h",
+		Minute:  "m",
+		Second:  "s",
+		MSecond: "ms",
+		USecond: "µs",
+		NSecond: "ns",
+	}
+
+	// We need to access the map in order using an ordered array
+	// Sort into iteration order: y, d, h, m, s, ...
+	var keys []Duration
+	for k := range units {
+		keys = append(keys, k)
+	}
+	sort.Slice(keys, func(i, j int) bool { return keys[i] > keys[j] })
+
+	var render string
+
+	// Invalid case: min > max
+	if min > max {
+		return "inv!"
+	}
+
+	// Special case: zero duration
+	if dt <= 0 {
+		return "0" + units[min]
+	}
+
+	// Special case: render as "ages!"
+	if dt >= float64(math.MaxUint64) {
+		return "ages!"
```

Comment:
By the way, don't we need to translate this?

Reply:
I am happy to do this, but I would prefer if we break this into another ticket as I have used the scheme currently being used without translation as in FormatDuration(). Happy to make things better (i.e. Make quantity.go language aware).

Reply:
For sure, if the code was not language-aware before, it can be done in another branch.